src/main/java/io/cdap/plugin/gcp/spanner/source/SpannerSource.java

```diff
@@ -111,20 +111,27 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
       Schemas.validateFieldsMatch(schema, configuredSchema, collector);
       stageConfigurer.setOutputSchema(configuredSchema);
-    } catch (SpannerException e) {
+    } catch (IOException | SpannerException e) {
       // this is because spanner exception error message is not very user friendly. It contains class names and new
       // lines in the error message.
-      collector.addFailure("Unable to connect to spanner instance.",
-                           "Verify spanner configurations such as instance, database, table, project, etc.")
-        .withStacktrace(e.getStackTrace());
+      // We don't want to fail pipelines which fail to connect in deployment
+      LOG.warn("Unable to connect to spanner instance with error:\n", e);
     }
   }
 
   @Override
   public void prepareRun(BatchSourceContext batchSourceContext) throws Exception {
     FailureCollector collector = batchSourceContext.getFailureCollector();
     config.validate(collector);
-    Schema actualSchema = getSchema(collector);
+    Schema actualSchema = null;
+    try {
+      actualSchema = getSchema(collector);
+    } catch (IOException e) {
+      collector.addFailure("Unable to get Spanner Client: " + e.getMessage(), null)
```

Comment:
this message doesn't seem correct (existing problem), as the exception can happen for many reasons. Better to say something more generic, like 'Unable to fetch schema of Spanner table %s' and include the table id in the message.